### PR TITLE
Msbuild retry

### PR DESCRIPTION
**Pull request by JelleHissink**
Retry of PR: 8

Now based on the new files option

### CODE REVIEW

File: src/LibSassBuilder/FilesOptions.cs [hunk 1]
```diff
@@ -0,0 +1,15 @@
+﻿using CommandLine;
+using System.Collections.Generic;
+
+namespace LibSassBuilder
+{
+	[Verb("files")]
+	public class FilesOptions : GenericOptions
+	{
+		[Value(0, Required = true, HelpText = "Directory in which to run. Defaults to current directory.")]
```
[the remark this answers is not in the document]
Comment (JelleHissink) [reply src/LibSassBuilder/FilesOptions.cs:9]: Because that other commit had not been merged yet...

File: package/build/LibSassBuilder.props [hunk 2]
```diff
@@ -2,6 +2,21 @@
   <!-- https://natemcmaster.com/blog/2017/11/11/build-tools-in-nuget/ -->
   <PropertyGroup>
     <SassExe Condition=" '$(SassExe)'=='' ">$(MSBuildThisFileDirectory)../tool/LibSassBuilder.dll</SassExe>
-    <LibSassBuilderArgs Condition=" '$(LibSassBuilderArgs)'=='' "></LibSassBuilderArgs>
```
[the remark this answers is not in the document]
Comment (JelleHissink) [reply package/build/LibSassBuilder.props:5]: I'll try to incorporate LibSassBuilderArgs

[the remark this answers is not in the document]
Comment (JelleHissink) [reply package/build/LibSassBuilder.props:5]: Done... ;-)

**Comment by johan-v-r** [comment]
Something's not working properly, or my understanding could be wrong...

Below my test findings with `dotnet watch build`

## Watch
`.scss` files aren't watched by default, I still had to explicitly add `<Watch Include="**\*.scss" Exclude="**\*.css" />`

## Files
After assigning the watch and logging (`<LibSassOutputLevel>verbose</LibSassOutputLevel>`) - then editing one of the `.scss` files, it seems to process all the `.scss` files:
```
watch : Waiting for a file to change before restarting dotnet...
watch : Started
Microsoft (R) Build Engine version 16.8.3+39993bd9d for .NET
Copyright (C) Microsoft Corporation. All rights reserved.

  Determining projects to restore...
  All projects are up-to-date for restore.
  Sass compile files
  Processing: C:\SourceCode\LibSassBuilderNugetTester\LibSassBuilderNugetTester\Pages\Counter.razor.scss
  Processing: C:\SourceCode\LibSassBuilderNugetTester\LibSassBuilderNugetTester\Shared\MainLayout.razor.scss
  Processing: C:\SourceCode\LibSassBuilderNugetTester\LibSassBuilderNugetTester\wwwroot\foo.scss
  Sass files compiled
```

My understanding was that MSBuild would now only send _changed_ files to LSB

**Comment by JelleHissink** [comment]
# Watch

If you specify a LibSassBuilderArgs SassFiles are ignored (https://github.com/johan-v-r/LibSassBuilder/pull/12/files#diff-540629219a4a03215aa6df95d26b5d60f458b7f9e4737b3bfa5c826716965046R18).
Because SassFile items are not used at all when you specify args. Would you still want <Watch> to work in that case?

# Files

It does not only send changed files, I did not know what files are included, referenced or used by what other files. But it would be a nice future addition. But then we would need a list of dependencies or rules (e.g. files starting with _ => trigger all?)
Also for now there is not a hash per file, but one hash over the filelisting/modification date.

In short for now: scss changes are detected the tool is run, otherwise not.

**Comment by johan-v-r** [comment]
## Watch

Nope that test was without specifying `LibSassBuilderArgs` - using just the default config with a local build from your branch, see my `.csproj`:
```xml
<Project Sdk="Microsoft.NET.Sdk.BlazorWebAssembly">

  <PropertyGroup>
    <TargetFramework>net5.0</TargetFramework>
  </PropertyGroup>

  <ItemGroup>
    <PackageReference Include="LibSassBuilder" Version="1.5.1" />
    <PackageReference Include="Microsoft.AspNetCore.Components.WebAssembly" Version="5.0.0" />
    <PackageReference Include="Microsoft.AspNetCore.Components.WebAssembly.DevServer" Version="5.0.0" PrivateAssets="all" />
    <PackageReference Include="System.Net.Http.Json" Version="5.0.0" />
  </ItemGroup>

  <PropertyGroup>
	<LibSassOutputLevel>verbose</LibSassOutputLevel>
  </PropertyGroup>

</Project>
```

## Files

So what's the difference/benefit of having MSBuild traversing the directory to find the files, vs letting LSB do that by default..?

**Comment by JelleHissink** [comment]
# Watch

I'll look into it, did not test after latest refactoring

# Files

* Msbuild already traversed it
* Same input for watch as for the tool
* Hash is only used to prevent calling the tool every compile. Might not be a great impact, but every second counts.
* Msbuild patterns/globing are more powerful, and properties / items are more user-friendly

**Comment by JelleHissink** [comment]
# Watch

I'll check... have not re-tested that part yet.

# Files

* msbuild has a really powerful globing system
* msbuild project structure and options are known, more easy to discover, ties in more with the exiting pattern.
* msbuild/vs can cache item lists in the project system
* Calling lsb is a small overhead, generating all files, and reading all files to test if it had changed was not needed, msbuild hash is only used to prevent this.

**Comment by JelleHissink** [comment]
For me dotnet watch now works again, had to move the property to the .props file for it to work? Strange... but ok I guess
